jobsdb/jobsdb.go

```diff
@@ -3328,30 +3490,28 @@ func (jd *HandleT) backupTable(ctx context.Context, backupDSRange *dataSetRangeT
 		return false, fmt.Errorf("creating gz file %q: %w", path, err)
 	}
 	defer os.Remove(path)
+	var offset int64
 
-	var offset, batchCount int64
 	for {
 		stmt := jd.getBackUpQuery(backupDSRange, isJobStatusTable, offset)
 		var rawJSONRows json.RawMessage
-		row := jd.dbHandle.QueryRow(stmt)
-		err = row.Scan(&rawJSONRows)
-		if err != nil {
-			panic(fmt.Errorf("Scanning row failed with error : %w", err))
-		}
-
-		rowEndPatternMatchCount += int64(bytes.Count(rawJSONRows, []byte("}, \n {")))
-		rawJSONRows = bytes.Replace(rawJSONRows, []byte("}, \n {"), []byte("}\n{"), -1) // replacing ", \n " with "\n"
-		batchCount++
-
-		// Asserting that the first character is '[' and last character is ']'
-		jd.assert(rawJSONRows[0] == byte('[') && rawJSONRows[len(rawJSONRows)-1] == byte(']'), "json agg output is not in the expected format. Excepted format: JSON Array [{}]")
-		rawJSONRows = rawJSONRows[1 : len(rawJSONRows)-1] // stripping starting '[' and ending ']'
-		rawJSONRows = append(rawJSONRows, '\n')           // appending '\n'
+		rows, err := jd.dbHandle.Query(stmt)
+		jd.assertError(err)
+		defer rows.Close()
 
-		if _, err := gzWriter.Write(rawJSONRows); err != nil {
-			return false, fmt.Errorf("writing to gz file %q: %w", path, err)
+		for rows.Next() {
+			err = rows.Scan(&rawJSONRows)
+			if err != nil {
+				panic(fmt.Errorf("Scanning row failed with error : %w", err))
+			}
+			rawJSONRows = append(rawJSONRows, '\n') // appending '\n'
```

Comment:
`[very minor]` not sure about the benefit of this comment
```suggestion
			rawJSONRows = append(rawJSONRows, '\n') 
```